src/partition/repart.c

```diff
@@ -1396,13 +1403,16 @@ static int context_read_definitions(
         _cleanup_strv_free_ char **files = NULL;
         Partition *last = NULL;
         int r;
+        const char *const *dirs;
 
         assert(context);
 
         if (directory)
-                r = conf_files_list_strv(&files, ".conf", NULL, CONF_FILES_REGULAR|CONF_FILES_FILTER_MASKED, (const char**) STRV_MAKE(directory));
+                dirs = STRV_MAKE_CONST(directory);
         else
-                r = conf_files_list_strv(&files, ".conf", root, CONF_FILES_REGULAR|CONF_FILES_FILTER_MASKED, (const char**) CONF_PATHS_STRV("repart.d"));
+                dirs = (const char* const*) CONF_PATHS_STRV("repart.d");
```

Comment:
You cannot use `dirs` outside of the scope, like `alloca()`.

Reply:
Thanks for spotting the issue! I will try your suggestion.

Reply:
What about using a ternary instead of an if block? That would also fix the issue.

Reply:
Good idea. Thanks for the suggestion.

Reply:
Good idea. Thanks for the suggestion.